### PR TITLE
Add tracer data reference and tracer input data API functions

### PR DESCRIPTION
**Pull request by tarcisiofischer**
ASIM-2266

### CODE REVIEW

File: alfasim_sdk_api/detail/api_pointers.h [hunk 1]
```diff
@@ -59,6 +59,24 @@ get_plugin_input_data_string_func get_plugin_input_data_string;
...
+*   get_plugin_input_data_reference
+*
+*   Get an ALFAsim input data internal Reference. Note that a Reference is a specific concept of
+*   ALFAsim SDK and plugins - It is a way to retrieve data from an input outside of the current
+*   model. See the ALFAsim's SDK python configuration file for more information.
+*
+*   Example usage:
+*
+*   errcode = get_plugin_input_data_reference(
+*       ctx, &tracer_ref, plugin_name, "Model.tracer_reference");
```
Comment (nicoddemus) [inline alfasim_sdk_api/detail/api_pointers.h]: It is useful to declare `tracer_ref` here in the example as well

File: alfasim_sdk_api/detail/api_pointers.h [hunk 2]
```diff
@@ -239,4 +257,46 @@ typedef int (*get_flow_pattern_func)(
 get_flow_pattern_func get_flow_pattern;
 
 
+/**
+*   get_tracer_id
+*
+*   Retrieves the tracer ID given a tracer reference. A tracer reference may be obtained from the
+*   user input data (See get_plugin_input_data_reference API function for an example).
```
Comment (nicoddemus) [inline alfasim_sdk_api/detail/api_pointers.h]: 👍

File: alfasim_sdk_api/detail/api_pointers.h [hunk 3]
```diff
@@ -239,4 +257,46 @@ typedef int (*get_flow_pattern_func)(
...
+*
+*   Retrieves the size of the tracer name, given a tracer reference. A tracer reference may be
+*   obtained from the user input data (See get_plugin_input_data_reference API function for an
+*   example).
+*/
+typedef int (*get_tracer_name_size_func)(void* ctx, int* tracer_name_size, void* reference);
+get_tracer_name_size_func get_tracer_name_size;
+
+
+/**
```
Comment (nicoddemus) [inline alfasim_sdk_api/detail/api_pointers.h]: Duplicated with the above

Comment (tarcisiofischer) [reply alfasim_sdk_api/detail/api_pointers.h]: Oops, forgot to update the docs

File: alfasim_sdk_api/detail/api_pointers.h [hunk 4]
```diff
@@ -239,4 +258,56 @@ typedef int (*get_flow_pattern_func)(
...
+*   get_tracer_name
+*
+*   Retrieves the tracer name, given a tracer reference. The tracer_name parameter must be a valid
+*   and pre-allocated memory region where the name string will be copied to. A tracer reference may
+*   be obtained from the user input data (See get_plugin_input_data_reference API function for an
+*   example).
+*
+*   Example usage:
+*   int tracer_name_size = -1;
+*   char tracer_name[100];
```
Comment (nicoddemus) [inline alfasim_sdk_api/detail/api_pointers.h]: bad practice, you should dynamically allocate the buffer using `tracer_name_size` 😉

**Pull request merged**